turbinia/turbiniactl.py

```diff
@@ -936,42 +931,32 @@ def process_evidence(
                   credential, args.decryption_keys, exception))
           raise TurbiniaException(msg)
         evidence_.credentials.append((credential_type, credential_data))
-    if args.recipe or args.recipe_path:
-      if (args.jobs_denylist or args.jobs_allowlist or
-          args.filter_patterns_file or args.yara_rules_file):
-        msg = (
-            'Specifying a recipe is incompatible with defining '
-            'jobs allow/deny lists, yara rules or a patterns file separately.')
-        raise TurbiniaException(msg)
 
-      if args.recipe_path:
-        recipe_file = args.recipe_path
-      else:
-        recipe_file = os.path.join(config.RECIPE_FILE_DIR, args.recipe)
-      if not os.path.exists(recipe_file) and not recipe_file.endswith('.yaml'):
-        log.warning(
-            'Could not find recipe file at {0:s}, checking for file '
-            'with .yaml extension'.format(recipe_file))
-        recipe_file = recipe_file + '.yaml'
-      if not os.path.exists(recipe_file):
-        msg = 'Recipe file {0:s} could not be found. Exiting.'.format(
-            recipe_file)
-        raise TurbiniaException(msg)
+    # Recipe pre-condition checks.
+    recipe_helpers.validate_recipe_conditions(args)
+    if not hasattr(args, 'skip_recipe_validation'):
+      skip_recipe_validation = False
+    else:
+      skip_recipe_validation = args.skip_recipe_validation
 
-      recipe_dict = recipe_helpers.load_recipe_from_file(
-          recipe_file, not args.skip_recipe_validation)
-      if not recipe_dict:
-        msg = 'Could not initiate a recipe_dict.'
-        raise TurbiniaException(msg)
+    if args.recipe or args.recipe_path:
+      # Load the specified recipe.
+      recipe_dict = client.create_recipe(
+          debug_tasks=args.debug_tasks, filter_patterns=filter_patterns,
+          group_id=group_id, jobs_allowlist=args.jobs_allowlist,
+          jobs_denylist=args.jobs_denylist,
+          recipe_name=args.recipe if args.recipe else args.recipe_path,
+          sketch_id=None, skip_recipe_validation=skip_recipe_validation,
+          yara_rules=yara_rules)
+      request.recipe = recipe_dict
     else:
-      recipe_dict = copy.deepcopy(recipe_helpers.DEFAULT_RECIPE)
-      recipe_dict['globals']['debug_tasks'] = args.debug_tasks
-      recipe_dict['globals']['filter_patterns'] = filter_patterns
-      recipe_dict['globals']['jobs_denylist'] = args.jobs_denylist
-      recipe_dict['globals']['jobs_allowlist'] = args.jobs_allowlist
-      recipe_dict['globals']['yara_rules'] = yara_rules
-    recipe_dict['globals']['group_id'] = group_id
-    request.recipe = recipe_dict
+      # Create a 'default' recipe.
```

Comment:
If we create a default recipe by default in `create_request()` then I don't think we'd need to have this second case here and would probably only need one call to `create_recipe()` here.

Reply:
Good suggestion, I went ahead and made that change.